### DISCUSSION TITLE
Allow Adding New Field List Values On-the-Fly in Article Editor

### DISCUSSION THREAD

**Discussion by keep** [joomla/joomla-cms#47032] [2. Cold Features]
### Steps to reproduce the issue

- Add a new custom field called "color" to com_content.article with the field type set to "list."
- Add the list values: red, white, and green.
- When editing an article, under the 'Fields' tab, choose one or more colors from the predefined list (depending on whether the field is set to allow multiple selections).
- Notice that if you need to add a new color (such as black, blue, or yellow) while editing the article, you must save and close the article, go to the field editor, add the new value, save the field, and then return to the article editor to choose the new color.

### Expected result
While editing an article, when there is no desired option in the "color" list, there should be a way to add a new value (e.g., black, blue, or yellow) directly from the article editor, such as an input field or button called 'New List Value.' The new value should be added on-the-fly, perhaps via AJAX or during the article save process.

### Actual result
There is currently no way to add new values to the "color" list directly from the article editor. Instead, the user must exit the article editor, navigate to the field editor, add the new values manually, and then return to the article editor to select the new value.

### System information (as much as possible)
Joomla 5.1.4, PHP 8.3

### Additional comments
This feature would improve workflow efficiency for users managing dynamic lists, reducing the need to leave the article editor when new options are required. It would be great if this functionality could be implemented via AJAX or another method that allows seamless value additions without needing to leave the current editing context.

**Comment by brianteeman** [comment]
Do you mean that you want a custom field that works in the same way as the current tags field?

**Comment by keep** [comment]
Yes, this would be great.

I’m not sure if you are familiar with the Custom Fields For All plugin from breakdesigns.net, which is a VirtueMart extension. It includes a "new value" button next to the selector, where you can add your new option.
You can check a screenshot here: https://breakdesigns.net/extensions/joomla/custom-fields-for-all<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44074">issues.joomla.org/tracker/joomla-cms/44074</a>.</sub>

**Comment by mentalhacker** [comment]
This feature is a 'must have' especially in the case of big news portals or large webshops where huge amount of data is stored in single or multiple select list type custom fields. If there were no need to leave the article editor when new options are required, it would significantly quicken the workflow of the editing process.
We have been lacking this function for ages.
Thanks for drawing the developers' attention to this issue.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44074">issues.joomla.org/tracker/joomla-cms/44074</a>.</sub>
